src/common/modify-settings-modal/index.tsx

```diff
@@ -322,7 +323,13 @@ const ModifySettingsModal = ({ position, open, onCancel }: ModifySettingsModalPr
           </Typography>
         ),
       });
-      const result = await walletService.approveToken(fromToUse, fromHasYield, version);
+
+      const result = await walletService.approveToken(
+        fromToUse,
+        fromHasYield,
+        version,
+        amount ? parseUnits(amount, from.decimals) : undefined
```

Comment:
This is not correct, they only need to approve the difference between what is already on the position and what they actually want the position to have in the end

`remainingLiquidity.sub(BigNumber.from(frequencyValue).mul(parseUnits(rate, fromToUse.decimals))).abs(),`